### PR TITLE
Add a doc for end to end acknowledgements

### PR DESCRIPTION
**Pull request by kkondaka**
### Description
Document describing the end to end acknowledgment support in Data Prepper
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
  - [X ] New functionality has javadoc added
- [ X] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Naarcha-AWS** [comment]
@kkondaka: Are you ok if I carry this over to the documentation-website repo and then tag you for review?

**Comment by kkondaka** [comment]
> @kkondaka: Are you ok if I carry this over to the documentation-website repo and then tag you for review?

Sure.

**Review by dlvenable — COMMENTED** [review]
You have some changes that were already made in `main`. You might have branched off your other branch instead of `main`.

File: docs/end_to_end_acknowledgements.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+### Limitations
+The end-to-end acknowledgement framework loses track of events sent to a different DataPrepper instance running on a different node, making it unsuitable for use when [PeerForwarder](https://github.com/opensearch-project/data-prepper/tree/main/docs/peer_forwarder.md) is used.
+As of DataPrepepr version 2.2, only S3 Source and OpenSearch Sink support end-to-end acknowledgements. End to end acknowledgements can be enabled for S3 source using `acknowledgements: true` config option.
```
Comment (dlvenable) [inline docs/end_to_end_acknowledgements.md:7]: typo: "DataPrepepr"

File: docs/end_to_end_acknowledgements.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
+
+## End To End Acknowledgements
```
Comment (dlvenable) [inline docs/end_to_end_acknowledgements.md:2]: Using "end to end" here is an adjective, so I believe you should use hyphens.

"End-to-end acknowledgements"

Please use that throughout.

Also, the OpenSearch documentation page is capitalizing only the first letter in a header for common nouns. So, this header should be:

```
## End-to-end acknowledgements
```

File: docs/end_to_end_acknowledgements.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+The Data Prepper framework offers a reliable way for sources to ensure the durability and reliability of data by tracking its delivery to sinks through end-to-end acknowledgements. If the pipeline has multiple sinks or multiple sub-pipelines configured, the event level acknowledgements are sent only after the event is sent to all sinks in all the sub-pipelines. The framework enables a source to create an acknowledgement set to monitor a batch of events and receive a positive acknowledgement when those events are successfully pushed to the sinks or negative acknowledgement when any one of the events could not be pushed to the sinks for any reason. In the event of a failure or crash of any component of the Data Prepper, if it is unable to send an event, the source will not receive an acknowledgement. In this case, the source will time out and can take necessary actions like retrying or logging the failure.
```
Comment (dlvenable) [inline docs/end_to_end_acknowledgements.md:3]: "multiple sub-pipelines configured"

I'm not sure that sub-pipelines is quite the right term here. They are destination pipelines, not necessarily sub-pipelines.

Perhaps:

"If the pipeline has multiple sinks, including other pipeline sinks, ..."

**Review by dlvenable — CHANGES_REQUESTED** [review]
We should also get some of this documentation into the opensearch.org documentation. The development-oriented documentation should be only here. But, the documentation for end-users should be available there as well.

File: docs/end_to_end_acknowledgements.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
+
+## End-to-end Acknowledgements
```
Comment (dlvenable) [inline docs/end_to_end_acknowledgements.md:2]: I think we are settling on no "e", so this should be `acknowledgments`.

**Review by dlvenable — APPROVED** [review]

File: docs/end_to_end_acknowledgements.md (hunk 4 above)
Comment (kkondaka) [reply docs/end_to_end_acknowledgements.md:2]: Yeah. That's only for external facing (config) word. I am being little flexible on general usage of the word.

**Review by asifsmohammed — APPROVED** [review]

**Pull request merged**